Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/VehicleRegistrationCertificate.hs

```diff
@@ -202,19 +230,13 @@ onVerifyRC verificationReq output = do
       mEncryptedRC <- encrypt `mapM` output.registration_number
       let mbFitnessEpiry = convertTextToUTC output.fitness_upto <|> Just (DT.UTCTime (TO.fromOrdinalDate 1900 1) 0)
       let mVehicleRC = createRC rCConfigs rCInsuranceConfigs output id verificationReq.documentImageId1 now <$> mEncryptedRC <*> mbFitnessEpiry
-
       case mVehicleRC of
         Just vehicleRC -> do
           runTransaction $ RCQuery.upsert vehicleRC
-
           -- linking to driver
           rc <- RCQuery.findByRCAndExpiry vehicleRC.certificateNumber vehicleRC.fitnessExpiry >>= fromMaybeM (InternalError "RC not found")
-          mRCAssociation <- DAQuery.getActiveAssociationByRC rc.id
-          when (isNothing mRCAssociation) $ do
-            currAssoc <- DAQuery.getActiveAssociationByDriver person.id
-            when (isJust currAssoc) $ do runTransaction $ DAQuery.endAssociation person.id
-            driverRCAssoc <- mkAssociation person.id rc.id
-            runTransaction $ DAQuery.create driverRCAssoc
+          driverRCAssoc <- mkAssociation person.id rc.id
```

Comment:
check if driver is not already linked to this RC. If not, then only create entry here.

Reply:
> check if driver is not already linked to this RC. If not, then only create entry here.
if a driver is already linked to same RC then we are throwing error before this that same RC already Linked.